### PR TITLE
fix #99

### LINKED ISSUE #99: support of '--tmp-ssh-key' option
- `scw run --tmp-ssh-key ...`
- `scw create --tmp-ssh-key ...`

### PR DESCRIPTION
**Pull request by QuentinPerez**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -309,6 +309,7 @@ Options:
   --bootscript=""       Assign a bootscript
   -e, --env=""          Provide metadata tags passed to initrd (i.e., boot=resue INITRD_DEBUG=1)
   -h, --help=false      Print usage
+  --local-ssh-key=false Creates an ssh key that you can use to connect on the image
```
Comment (moul) [inline README.md]: the main goal of this function is not to create a local ssh key, but to use a local ssh key, and create a new one if no one was found, try to reformulate the description to keep the priority of the features

File: README.md [hunk 2]
```diff
@@ -628,6 +630,7 @@ Options:
   -e, --env=""          Provide metadata tags passed to initrd (i.e., boot=resue INITRD_DEBUG=1)
   -g, --gateway=""      Use a SSH gateway
   -h, --help=false      Print usage
+  --local-ssh-key=false Creates an ssh key that you can use to connect on the image
```
Comment (moul) [inline README.md]: https://github.com/scaleway/scaleway-cli/pull/125/files#r37165002

File: pkg/commands/run.go [hunk 3]
```diff
@@ -11,30 +11,82 @@ import (
...
+	LocalSSHKey bool
 	// DynamicIPRequired
 	// Timeout
 }
 
+// AddSSHKeyToTags adds the ssh key in the tags
+func AddSSHKeyToTags(ctx CommandContext, tags *[]string, image string) error {
+	home, err := utils.GetHomeDir()
+	if err != nil {
+		return fmt.Errorf("Unable to find your home %v", err)
```
Comment (moul) [inline pkg/commands/run.go]: return fmt.Errorf needs to begin with a lowercase word

File: pkg/commands/run.go [hunk 4]
```diff
@@ -25,16 +28,53 @@ type RunArgs struct {
...
+	home, err := utils.GetHomeDir()
+	if err != nil {
+		return fmt.Errorf("unable to find your home %v", err)
+	}
+	idRsa := filepath.Join(home, ".ssh", "id_rsa")
+	if _, err := os.Stat(idRsa); err != nil {
+		if os.IsNotExist(err) {
+			logrus.Warnln("Unable to find your ~/.ssh/id_rsa")
+			logrus.Warnln("Run 'ssh-keygen -t rsa'")
+			return nil
```
Comment (moul) [inline pkg/commands/run.go]: By returning nil, the command main command (run/create) won't fail

File: pkg/commands/run.go [hunk 5]
```diff
@@ -11,6 +11,9 @@ import (
 	"github.com/scaleway/scaleway-cli/pkg/api"
 	"github.com/scaleway/scaleway-cli/pkg/utils"
 	"github.com/scaleway/scaleway-cli/vendor/github.com/Sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 )
```
Comment (moul) [inline pkg/commands/run.go]: Are you using [goimports](https://github.com/bradfitz/goimports) ?

File: pkg/commands/create.go [hunk 6]
```diff
@@ -18,10 +18,18 @@ type CreateArgs struct {
 	Tags       []string
 	Volumes    []string
 	Image      string
+	TmpSSHKey  bool
 }
 
 // RunCreate is the handler for 'scw create'
 func RunCreate(ctx CommandContext, args CreateArgs) error {
+	if args.TmpSSHKey == true {
```
Comment (moul) [inline pkg/commands/create.go]: Not sure, but I think golint prefer `if args.TmpSSHKey {` without `== true`

File: pkg/commands/run.go (hunk 4 above)
Comment (QuentinPerez) [inline pkg/commands/run.go]: It's just a warning, the image can be run/create without ~/.ssh/id_rsa

Comment (moul) [inline pkg/commands/run.go]: ok, go go go

File: pkg/commands/create.go (hunk 6 above)
Comment (QuentinPerez) [inline pkg/commands/create.go]: $> golint pkg/commands/create.go
$>

I have no error, but I can change if you want 😊

Comment (moul) [inline pkg/commands/create.go]: nop, that's good

File: pkg/commands/run.go (hunk 5 above)
Comment (QuentinPerez) [inline pkg/commands/run.go]: Oups I used gofmt

**Comment by moul** [comment]
LGTM

**Pull request merged**